### PR TITLE
Convert extra when extra is not serializable

### LINKED ISSUE #180: Catch exception during the creation of JsonPacket
The code [here](https://github.com/getsentry/raven-csharp/blob/develop/src/app/SharpRaven/RavenClient.cs#L224) may throw exceptions. I think we should catch it and leave it to `HandleException`.

### PR DESCRIPTION
**Pull request by skyline75489**
This fixed #180

### CODE REVIEW

File: src/app/SharpRaven/Data/JsonPacket.cs [hunk 1]
```diff
@@ -123,7 +123,16 @@ private static object Merge(SentryEvent @event)
                 }
             }
             else
-                result = JObject.FromObject(extra);
+            {
+                try
+                {
+                    result = JObject.FromObject(extra);
+                }
+                catch (ArgumentException e)
```
Comment (asbjornu) [inline src/app/SharpRaven/Data/JsonPacket.cs]: No need for an instance `e` here, is it?

File: src/app/SharpRaven/Data/JsonPacket.cs [hunk 2]
```diff
@@ -123,7 +123,16 @@ private static object Merge(SentryEvent @event)
...
+                {
+                    result = JObject.Parse(string.Format(@"{{""{0}"":""{1}""}}", extra.GetType(), extra));
```
Comment (asbjornu) [inline src/app/SharpRaven/Data/JsonPacket.cs]: This seems a bit sub-optimal. Can't we instantiate a new `JObject` and give it a property somehow?

File: src/app/SharpRaven/Data/JsonPacket.cs (hunk 1 above)
Comment (skyline75489) [inline src/app/SharpRaven/Data/JsonPacket.cs]: Indeed.

File: src/app/SharpRaven/Data/JsonPacket.cs (hunk 2 above)
Comment (skyline75489) [inline src/app/SharpRaven/Data/JsonPacket.cs]: It seems the solution is either this or `dynamic` object. I personally do not favor any usage of `dynamic`, though.

Comment (asbjornu) [inline src/app/SharpRaven/Data/JsonPacket.cs]: Seems like this works just fine:

```c#
result = JObject.FromObject(new Dictionary<string, object>
{
    { extra.GetType().ToString(), extra }
});
```

Comment (skyline75489) [inline src/app/SharpRaven/Data/JsonPacket.cs]: Yeah there's `Dictionary`... I'm too obsessed with the dynamic type thing in JSON and forgot about it. Should it be `Dictionary<string, string>`, though? Or it's handled by Json.Net itself?

Comment (asbjornu) [inline src/app/SharpRaven/Data/JsonPacket.cs]: JSON.NET serializes `object` just fine. 😃

**Comment by asbjornu** [comment]
@skyline75489: Excellent work, thanks!

**Pull request merged**